### PR TITLE
Fix for compatibility with Rails 5.1

### PR DESCRIPTION
**Pull request by hasghari**

### CODE REVIEW

File: lib/arel-helpers/join_association.rb [hunk 1]
```diff
@@ -108,7 +108,7 @@ def join_association_5_0(table, association, join_type)
         constraints = join_dependency.join_constraints([], join_type)
 
         binds = constraints.flat_map do |info|
-          prepared_binds = table.connection.prepare_binds_for_database(info.binds)
+          prepared_binds = info.binds.map(&:value_for_database)
```
Comment (hasghari) [inline lib/arel-helpers/join_association.rb]: @camertron This change comes from this PR https://github.com/rails/rails/pull/25937/files

**Comment by camertron** [comment]
Nice, thanks @hasghari :)

**Comment by camertron** [comment]
Published as v2.4.0.

**Comment by hasghari** [comment]
Awesome, thanks for the quick merge and release.

**Pull request merged**